File: static.c
#include <stdio.h>
void increase()
{
  // static 변수는 프로그램이 종료하기 전까지 변수가 사라지지 않고 계속 유지됨
  // increase 함수 호출 시 count는 static 변수이기 때문에 처음 한 번만 count 값이 0으로 초기화되고
  // increase 함수를 두 번 호출할 때 static int count = 0을 만나더라도 초기화되지 않음
  static int count = 0;

  printf("count=%d\n", count);
  count++;
}

int main(int argc, char *argv[])
{
  increase();
  increase();
  return 0;
}

File: Static.java
class Count {

  // static 변수는 한번만 초기화됨
  static int count = 0;
}

public class Static {

  public static void main(String[] args) {
    // static은 프로그램이 종료될 때까지 변수가 유지되고 최초 한번만 초기화 되기 때문에
    // 1, 2가 출력됨
    Count c = new Count();
    c.count++;
    System.out.println(c.count);
    c.count++;
    System.out.println(c.count);
  }
}
